service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/azure/relayNamespace/ControlledAzureRelayNamespaceResource.java

```diff
@@ -197,7 +197,7 @@ public void validate() {
       throw new MissingRequiredFieldException(
           "Missing required region field for ControlledAzureRelayNamespace.");
     }
-    ResourceValidationUtils.validateRegion(getRegion());
+    ResourceValidationUtils.validateAzureRegion(getRegion());
```

Comment:
So if this is here, why is it that we need it in the controller?

Reply:
it's not backward compatible for gcp resources which doesn't do region validation before this PR (e.g. if we stored an invalid region, we won't be able to deserialize it after this PR) not a problem for azure resources.

Reply:
How would we store an invalid region? We couldn't have successfully created the cloud resource if the region was invalid, right?

Good to think about the compatibility aspect. Is this a case where we need to do some data cleaning before we start enforcing the policy?

If you buy my logic that all of the regions have to be valid - if not properly down-cased and de-zoned - then we could do the cleaning on retrieval. Parse the region string and put it in canonical form. If the form changed, then update the database. Maybe??

Reply:
Interesting. I think your logic is right in that we shouldn't have an invalid region saved, though maybe not de-zoned. The call to GCP should have failed. So its flight should have too.
I'm not sure that will affect existing resources in the context of this MR now that we're validating during Controlled Resource Create. Existing resources have already been created. 
The cleaning existing resources to use canonical names. I think we need a follow-up task for that.

Reply:
Putting the resource cleanup into a separate ticket is fine with me. Are we likely to run into the same issue @yuhuyoyo is working on with buckets? That we can't enforce the validation when coming out of the database?

Reply:
yeah i think that sounds right. there might be zonal string in the db but not invalid ones.

Reply:
I don't think we'll hit an error like that here since we're only calling this validation on create. It should only apply to newly created items.

The next step is to add validation like this to clone. In that case we would be looking at the existing item's region, so we might need some zone => region normalizing there.